internal/pkg/database/challenge.go

```diff
@@ -0,0 +1,29 @@
+package database
+
+import (
+	"context"
+	"fmt"
+
+	"beem-auth/internal/pkg/util/random"
+
+	"github.com/google/uuid"
+)
+
+type Challenge struct {
```

Comment:
It looks like Challenges can be used for multiple things, but there's no way to specify what the challenge is for. Maybe rename to their specific use or enable more usecases for them?

Reply:
I think it's better to add that when we need it? We don't have a need for it now so it might be pointless to do it. I'm not even sure whether we will have other challenges in the future.